efficientnetv2/main_tf2.py

```diff
@@ -118,189 +92,321 @@ def test_step(self, data):
-          verbose=2 if strategy == 'tpu' else 1,
+          self.get_dataset(training=True, image_size=_image_size, config=self.config),
+          initial_epoch=start_epoch,
+          epochs=end_epoch,
+          **fit_config
       )
-    elif FLAGS.mode == 'train':
-      if not config.train.stages:
-        model.fit(
-            get_dataset(training=True, image_size=train_size, config=config),
-            epochs=config.train.epochs,
-            steps_per_epoch=steps_per_epoch,
-            callbacks=[ckpt_callback, tb_callback, rstr_callback],
-            verbose=2 if strategy == 'tpu' else 1,
-        )
-      else:
-        total_stages = config.train.stages
-        ibase = config.data.ibase or (train_size / 2)
-
-        if config.train.sched:
-          ram_list = np.linspace(5, config.data.ram, total_stages)
-          mixup_list = np.linspace(0, config.data.mixup_alpha, total_stages)
-          cutmix_list = np.linspace(0, config.data.cutmix_alpha, total_stages)
-
-        for stage in range(total_stages):
-          ratio = float(stage + 1) / float(total_stages)
-          start_epoch = int(
-              float(stage) / float(total_stages) * config.train.epochs)
-          end_epoch = int(ratio * config.train.epochs)
-          image_size = int(ibase + (train_size - ibase) * ratio)
-
-          if config.train.sched:
-            config.data.ram = ram_list[stage]
-            config.data.mixup_alpha = mixup_list[stage]
-            config.data.cutmix_alpha = cutmix_list[stage]
-
-          model.fit(
-              get_dataset(training=True, image_size=image_size, config=config),
-              initial_epoch=start_epoch,
-              epochs=end_epoch,
-              steps_per_epoch=steps_per_epoch,
-              callbacks=[ckpt_callback, tb_callback, rstr_callback],
-              verbose=2 if strategy == 'tpu' else 1,
-          )
-    elif FLAGS.mode == 'eval':
-      for ckpt in tf.train.checkpoints_iterator(
-          FLAGS.model_dir, timeout=60 * 60 * 24):
-        model.load_weights(ckpt)
-        eval_results = model.evaluate(
-            get_dataset(training=False, image_size=eval_size, config=config),
-            batch_size=config.eval.batch_size,
-            steps=num_eval_images // config.eval.batch_size,
-            callbacks=[tb_callback, rstr_callback],
-            verbose=2 if strategy == 'tpu' else 1,
-        )
-
-        try:
-          current_epoch = int(os.path.basename(ckpt).split('-')[1])
-        except IndexError:
-          logging.info('%s has no epoch info: stop!', ckpt)
-          break
-
-        logging.info('Epoch: %d, total %d', current_epoch, config.train.epochs)
-        if current_epoch >= config.train.epochs:
-          break
+
+  def evaluate(self) -> None:
+    """ Evaluation helper function. """
+    for ckpt in tf.train.checkpoints_iterator(
+        FLAGS.model_dir, timeout=60 * 60 * 24):
+      model.load_weights(ckpt)
+      eval_results = model.evaluate(
```

Comment:
@mingxingtan eval_results we are not printing?

Reply:
Oh, you can print it out.